### PR TITLE
Add service to restore TACACS from old config

### PR DESCRIPTION
**Pull request by renukamanavalan**
#### Why I did it
In upgrade scenarios, where config_db.json is not carry forwarded to new image, it could be left w/o TACACS credentials.
Added a service to trigger 5 minutes after boot and restore TACACS, if /etc/sonic/old_config/tacacs.json is present.

#### How I did it
By adding a service, that would fire 5 mins after boot.
This service apply tacacs if available.

#### How to verify it
Upgrade and watch status of tacacs.timer & tacacs.service
You may create /etc/sonic/old_config/tacacs.json, with updated credentials
(_before 5mins after boot_) and see that appears in config & persisted too.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [x] 201911
- [x] 202006
- [x] 202012

### CODE REVIEW

File: files/build_templates/sonic_debian_extension.j2 [hunk 1]
```diff
@@ -523,6 +523,13 @@ sudo cp $IMAGE_CONFIGS/config-setup/config-setup $FILESYSTEM_ROOT/usr/bin/config
 echo "config-setup.service" | sudo tee -a $GENERATED_SERVICE_FILE
 sudo LANG=C chroot $FILESYSTEM_ROOT systemctl enable config-setup.service
 
+# Add delayed tacacs application service
+sudo cp files/build_templates/tacacs.timer $FILESYSTEM_ROOT_USR_LIB_SYSTEMD_SYSTEM/
+echo "tacacs.timer" | sudo tee -a $GENERATED_SERVICE_FILE
+
+sudo cp files/build_templates/tacacs.service $FILESYSTEM_ROOT_USR_LIB_SYSTEMD_SYSTEM/
+echo "tacacs.service" | sudo tee -a $GENERATED_SERVICE_FILE
```
Comment (jleveque) [inline files/build_templates/sonic_debian_extension.j2:531]: I feel simply calling this service `tacacs` is a bit misleading. Maybe something like `tacacs-config` is more appropriate?

Comment (renukamanavalan) [reply files/build_templates/sonic_debian_extension.j2]: Agree. Done.

File: files/build_templates/sonic_debian_extension.j2 [hunk 2]
```diff
@@ -523,6 +523,13 @@ sudo cp $IMAGE_CONFIGS/config-setup/config-setup $FILESYSTEM_ROOT/usr/bin/config
 echo "config-setup.service" | sudo tee -a $GENERATED_SERVICE_FILE
 sudo LANG=C chroot $FILESYSTEM_ROOT systemctl enable config-setup.service
 
+# Add delayed tacacs application service
+sudo cp files/build_templates/tacacs_config.timer $FILESYSTEM_ROOT_USR_LIB_SYSTEMD_SYSTEM/
+echo "tacacs_config.timer" | sudo tee -a $GENERATED_SERVICE_FILE
```
Comment (jleveque) [inline files/build_templates/sonic_debian_extension.j2:528]: Suggest using hyphen instead of underscore (`tacacs-config`)

**Review by jleveque — CHANGES_REQUESTED** [review]

**Review by jleveque — APPROVED** [review]

**Comment by renukamanavalan** [comment]
```
admin@str-s6000-acs-13:~$ sudo systemctl status tacacs-config.timer
â— tacacs-config.timer - Delays tacacs apply until SONiC has started
     Loaded: loaded (/lib/systemd/system/tacacs-config.timer; enabled-runtime; vendor preset: enabled)
     Active: active (waiting) since Fri 2021-06-04 02:59:21 UTC; 4min 0s ago
    Trigger: Fri 2021-06-04 03:03:48 UTC; 25s left
   Triggers: â— tacacs-config.service

Jun 04 02:59:21 sonic systemd[1]: Started Delays tacacs apply until SONiC has started.
admin@str-s6000-acs-13:~$ sleep 15s
admin@str-s6000-acs-13:~$ sudo systemctl status tacacs-config.timer
â— tacacs-config.timer - Delays tacacs apply until SONiC has started
     Loaded: loaded (/lib/systemd/system/tacacs-config.timer; enabled-runtime; vendor preset: enabled)
     Active: active (running) since Fri 2021-06-04 02:59:21 UTC; 4min 38s ago
    Trigger: n/a
   Triggers: â— tacacs-config.service

Jun 04 02:59:21 sonic systemd[1]: Started Delays tacacs apply until SONiC has started.
admin@str-s6000-acs-13:~$ sudo systemctl status tacacs-config.service
â— tacacs-config.service - TACACS application
     Loaded: loaded (/lib/systemd/system/tacacs-config.service; static)
     Active: active (exited) since Fri 2021-06-04 03:04:07 UTC; 5s ago
TriggeredBy: â— tacacs-config.timer
    Process: 5793 ExecStart=/usr/bin/config-setup apply_tacacs (code=exited, status=0/SUCCESS)
   Main PID: 5793 (code=exited, status=0/SUCCESS)

Jun 04 03:04:00 str-s6000-acs-13 systemd[1]: Starting TACACS application...
Jun 04 03:04:01 str-s6000-acs-13 config-setup[5793]: Applied tacacs json to restore tacacs credentials
Jun 04 03:04:04 str-s6000-acs-13 config-setup[5795]: Running command: /usr/local/bin/sonic-cfggen -d --print-data 
Jun 04 03:04:07 str-s6000-acs-13 systemd[1]: Finished TACACS application.
admin@str-s6000-acs-13:~$ show tacacs
TACPLUS global auth_type login
TACPLUS global timeout 5 (default)
TACPLUS global passkey testing123

TACPLUS_SERVER address 100.127.20.21
               priority 1
               tcp_port 49
```

**Comment by abdosi** [comment]
@renukamanavalan please create PR for 201911. Sevice file changes `sonic.target` is not available in 201911.

**Comment by yxieca** [comment]
@renukamanavalan this PR cannot be cherry picked cleanly into 201811 branch. Please create separate PR for 201811 branch. Thanks!

**Pull request merged**